services/sentry/lib/getPayout.js

```diff
@@ -1,18 +1,71 @@
+/* eslint-disable no-nested-ternary */
+// const BN = require('bignumber.js') // allows
 const BN = require('bn.js')
 const toBalancesKey = require('../toBalancesKey')
 
 function getPayout(channel, ev) {
 	if (ev.type === 'IMPRESSION' && ev.publisher) {
 		// add the minimum price for the event to the current amount
-		return [toBalancesKey(ev.publisher), new BN(channel.spec.minPerImpression || 1)]
+		const minPrice = new BN(channel.spec.minPerImpression || 1)
+		const maxPrice = new BN(channel.spec.maxPerImpression || 1)
+		const price = channel.spec.priceMultiplicationRules
+			? payout(channel.spec.priceMultiplicationRules, ev, maxPrice, minPrice)
+			: new BN(channel.spec.minPerImpression || 1)
+		return [toBalancesKey(ev.publisher), new BN(price.toString())]
 	}
 	if (ev.type === 'CLICK' && ev.publisher) {
-		return [
-			toBalancesKey(ev.publisher),
-			new BN((channel.spec.pricingBounds && channel.spec.pricingBounds.CLICK.min) || 0)
-		]
+		const minPrice = new BN(
```

Comment:
I think you should make a convenience method `getBounds` which returns `[min, max]`, which gets it from pricingBounds and fallbacks for 0 (except for IMPRESSION, where it fallsback to min/maxPerImpression)